File: fastapi/entity/po/subCategory.py
from datetime import datetime
from typing import Optional

from sqlmodel import Field, SQLModel


class SubCategory(SQLModel, table=True):
    """子分类实体类"""

    __tablename__ = "sub_category"
    id: Optional[int] = Field(default=None, primary_key=True, index=True)
    name: str = Field()
    category_id: Optional[int] = Field(default=None)  # 所属分类ID
    create_time: Optional[datetime] = Field(default=None)
    update_time: Optional[datetime] = Field(default=None)

File: spring/src/main/java/com/hcsy/spring/entity/po/SubCategory.java
package com.hcsy.spring.entity.po;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
@Table("sub_category")
@Schema(description = "子分类")
public class SubCategory {
    @Schema(description = "子分类ID")
    @Id
    private Long id;

    @Schema(description = "子分类名称")
    private String name;

    @Schema(description = "父分类ID")
    private Long categoryId;

    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    private LocalDateTime updateTime;
}
